os/arch/arm/src/armv7-m/mpu.h

```diff
@@ -319,13 +312,15 @@ static inline void mpu_control(bool enable, bool hfnmiena, bool privdefena)
  ****************************************************************************/
 
 #if defined(CONFIG_ARMV7M_HAVE_ICACHE) || defined(CONFIG_ARMV7M_DCACHE)
-static inline void mpu_priv_stronglyordered(uintptr_t base, size_t size)
+static inline void mpu_priv_stronglyordered(uint32_t region, uintptr_t base, size_t size)
 {
-	unsigned int region = mpu_allocregion();
 	uint32_t regval;
 	uint8_t l2size;
 	uint8_t subregions;
 
+#if defined(CONFIG_ARMV7M_MPU_NREGIONS)
```

Comment:
In my opinion we dont need to check this. NREGION is always set to some value and we always need to verify that reion number that we pass is less than REGION. Please change in all the API's.

Reply:
changed